### ISSUE TITLE
missing crucial inferences, such as hippocampal neuron is-a forebrain neuron

### ISSUE THREAD

**Issue by cmungall** [obophenotype/cell-ontology#1337]
I was surprised to see these two terms are not in a subsumption relationship:

![image](https://user-images.githubusercontent.com/50745/144158528-bad48d52-eebf-4b74-be31-9a3b0dbc2b39.png)

The reason for this is that one term uses this pattern: https://obophenotype.github.io/cell-ontology/patterns/cellPartOfAnatomicalEntity/

the other users the has-soma-location pattern

this is effectively creating a **ragged lattice** where subsumption is blocked by being just-a-bit-too-specific-on-the-wrong axis

Some other issues I note here

the genus of both terms is a horrible sensu term

'CNS neuron (sensu Vertebrata)'
 and ('has soma location' some forebrain)

this is over-specification. There is a danger of ragged lattices here (also we are lacking TCs on the genus - I thought I added all these)

For reasoning to work to any degree we need to

 1 be much stricter about making terms follow consistent DPs.
 2 for very lax patterns, impose stricter exclusion criteria on genus terms
 3 have tools for calculating ragged lattices

2 and 3 are hard, but we have tools for 1. In mondo we auto-tag terms with the metaclass they instantiate using dc:conformsTo. We have have sparql rules that are applicable to some patterns.

E.g. if we want to use has-soma-location, we should have rules that neurons in the brain shouldn't be defined with the more general part-of

**Comment by dosumis** [comment]
> if we want to use has-soma-location, we should have rules that neurons in the brain shouldn't be defined with the more general part-of

Agreed.  We have some guidance here: https://github.com/obophenotype/cell-ontology/blob/master/docs/relations_guide.md#recording-location

@shawntanzk - I thought we'd updated all these.  Can you look into this & fix any that have been missed.

**Comment by shawntanzk** [comment]
> @shawntanzk - I thought we'd updated all these. Can you look into this & fix any that have been missed.

I fixed it very specifically - making MTG part of cerebral cortex 
Happy to look into this, will try to figure out what all the above means >.<

**Comment by shawntanzk** [comment]
@dosumis @cmungall just took a look at this, I can fix it for now by changing it to hippocampal neuron to to 'has soma location' some 'ammon's horn' and while I'm at it, change all the CNS neuron to use 'has soma location' instead of 'part of', but I do not know how to deal with the (sensu Vertebrata) over-specification issue

**Comment by shawntanzk** [comment]
The PR merged above will fix this issue, but wider question on querying is still there. @cmungall mentioned he will write up another ticket for this.
